insights/parsers/lspci.py

```diff
@@ -1,57 +1,134 @@
-    >>> len(pci_info.get("Network controller"))
-    1
-    >>> "Centrino Advanced-N 6205" in pci_info
-    True
-    >>> "0d:00.0" in pci_info
-    True
+Lists ALL the PCI Devices With Details
+======================================
+
+Parsers included in this module are:
+
+LsPci - command ``/sbin/lspci``
+-------------------------------
+
+LsPciDriver - command ``/sbin/lspci -k``
+----------------------------------------
+
 """
-from .. import LogFileOutput, parser, CommandParser
+
+from .. import LogFileOutput, parser, CommandParser, get_active_lines
 from insights.specs import Specs
 
 
+@parser(Specs.lspci_kernel)
+class LsPciDriver(CommandParser):
+    """
+    This module provides methods to access driver information of PCI devices from
+    the ``/sbin/lspci -k`` command.
+
+    Typical output of the ``lspci -k`` command is::
+
+        00:00.0 Host bridge: Intel Corporation 5500 I/O Hub to ESI Port (rev 13)
+                Subsystem: Cisco Systems Inc Device 0101
+        00:01.0 PCI bridge: Intel Corporation 5520/5500/X58 I/O Hub PCI Express Root Port 1 (rev 13)
+                Kernel driver in use: pcieport
+                Kernel modules: shpchp
+        00:02.0 PCI bridge: Intel Corporation 5520/5500/X58 I/O Hub PCI Express Root Port 2 (rev 13)
+                Kernel driver in use: pcieport
+                Kernel modules: shpchp
+        06:00.0 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)
+                Subsystem: Cisco Systems Inc Device 004a
+                Kernel driver in use: ixgbe
+                Kernel modules: ixgbe
+        06:00.1 Ethernet controller: Intel Corporation 82598EB 10-Gigabit AF Dual Port Network Connection (rev 01)
+                Subsystem: Cisco Systems Inc Device 004a
+                Kernel driver in use: ixgbe
+                Kernel modules: ixgbe
+
+    Example:
+
+        >>> lspcidriv = shared[LsPciDriver]
+        >>> pci_devs = lspcidriv.pci_dev_details.keys()
+        ['00:00.0', '00:01.0', '00:02.0', '06:00.0', '06:00.1']
+        >>> lspcidriv.get_pci_dev(pci_devs[0])
+        {'Subsystem': 'Cisco Systems Inc Device 0101', 'Dev_Details': 'Host bridge: Intel Corporation 5500 I/O Hub to ESI Port (rev 13)'}
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.pci_dev_details = {}
+        """dict: Dictionary service detail like active, running, exited, dead"""
+        super(LsPciDriver, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        """
+        Main parsing class method which stores all interesting data from the content.
+
+        Args:
+            content (context.content): Parser context content
+        """
+        for line in get_active_lines(content):
+            parts = line.split(None)
+            if parts[0] not in ['Subsystem:', 'Kernel']:
+                bus_device_function = parts[0]
```

Comment:
Better add a initialization for `bus_device_function` out of the if-clause, since it's used in the `else` branch too.